cpp/src/detail/utility_wrappers_impl.cuh

```diff
@@ -102,18 +104,55 @@ void transform_increment_ints(raft::device_span<value_t> values,
                     }));
 }
 
-template <typename new_vertex_t, typename old_vertex_t>
-void transform_cast_ints(raft::device_span<new_vertex_t> new_vertices,
-                         raft::device_span<old_vertex_t> old_vertices,
-                         rmm::cuda_stream_view const& stream_view)
+template <typename new_type_t>
+void copy_or_transform(raft::device_span<new_type_t> output,
+                       cugraph_type_erased_device_array_view_t const* input,
+                       rmm::cuda_stream_view const& stream_view)
 {
-  thrust::transform(rmm::exec_policy(stream_view),
-                    old_vertices.begin(),
-                    old_vertices.end(),
-                    new_vertices.begin(),
-                    cuda::proclaim_return_type<new_vertex_t>([] __device__(old_vertex_t value) {
-                      return static_cast<new_vertex_t>(value);
-                    }));
+  auto input_ =
+    reinterpret_cast<cugraph::c_api::cugraph_type_erased_device_array_view_t const*>(input);
```

Comment:
With the code in the c_api namespace, the reinterpret cast would not be necessary.